Standards/scs-01XX-v1-secure-connections.md

```diff
@@ -0,0 +1,178 @@
+
+### Transport Layer Security (TLS)
+
+All server-side TLS configurations integrated into the infrastructure as covered by this standard MUST adhere to the folllowing rules:
+
+- The following deprecated TLS versions MUST NOT be used:
+  - v1.0
+  - v1.1
+- The following cipher suites MUST NOT be used:
+  - any RC4 cipher suite
+  - any DES cipher suite
+  - any cipher suite involving MD5 hash algorithms
+  - any cipher suite with less than 128-bit encryption
+- Cipher suites utilizing CBC mode SHOULD NOT be used
+- Cipher suites with Perfect Forward Secrecy (using ephemeral keys with the 'E' suffix, e.g. DHE, ECDHE) SHOULD be used instead of their static key counterpart (e.g. DH, ECDH)
+
+### API Interfaces
+
+- Internal API endpoints of all OpenStack services MUST use TLS. Their endpoint as registered in the Keystone service catalog MUST be an HTTPS address.
+- External API endpoints of all OpenStack services MUST use TLS. Their endpoint as registered in the Keystone service catalog MUST be an HTTPS address.
+
+You MAY refer to [TLS proxies and HTTP services](https://docs.openstack.org/security-guide/secure-communication/tls-proxies-and-http-services.html) and [Secure reference architectures](https://docs.openstack.org/security-guide/secure-communication/secure-reference-architectures.html) of the OpenStack Security Guide for best practices and recommendations.
+
+### Database Connections
+
+- The database servers used by the OpenStack services MUST be configured for TLS transport.
+- All OpenStack services MUST have TLS configured for the database connection via the `ssl_ca` directive. See [OpenStack service database configuration](https://docs.openstack.org/security-guide/databases/database-access-control.html#openstack-service-database-configuration).
+- Database user accounts for the OpenStack services SHOULD be configured to require TLS connections via the `REQUIRE SSL` SQL directive. See [Require user accounts to require SSL transport](https://docs.openstack.org/security-guide/databases/database-access-control.html#require-user-accounts-to-require-ssl-transport).
+- Security MAY be further enhanced by configuring the OpenStack services to use X.509 client certificates for database authentication. See [Authentication with X.509 certificates](https://docs.openstack.org/security-guide/databases/database-access-control.html#authentication-with-x-509-certificates).
+
+### Message Queue Connections
+
+- If using RabbitMQ or Qpid as the message queue service, the SSL functionality of the message broker MUST be enabled and used by the OpenStack services. See [Messaging transport security](https://docs.openstack.org/security-guide/messaging/security.html#messaging-transport-security).
+  - If using RabbitMQ, the OpenStack services' oslo.messaging configuration for RabbitMQ MUST use the `kombu_ssl_*` options accordingly to enable SSL.
```

Comment:
The `kombu_ssl` names are long [deprecated](https://docs.openstack.org/oslo.messaging/latest/configuration/opts.html#oslo_messaging_rabbit.ssl_key_file). I don't know anybody using those old deprecated names. Also I think a new standard should use the upstreams standard name, if it must use any name.

```suggestion
  - If using RabbitMQ, the OpenStack services' oslo.messaging configuration for RabbitMQ MUST use the `ssl_*` options accordingly to enable SSL.
```
But this is very unspecific, still.

A complete example looks like this:

```
[oslo_messaging_rabbit]
ssl = true # not catched by above regex, strictly speaking
ssl_ca_file = /path/to/file
ssl_key_file = 
ssl_cert_file = 
```

you can enable this globally e.g. in kolla via the guide:
https://docs.openstack.org/kolla-ansible/latest/reference/message-queues/rabbitmq.html#tls-encryption

`rabbitmq_enable_tls = true`

But, to quote the same page, not everything is encrypted currently:

```
Encryption of the following channels is not currently supported:

    RabbitMQ cluster traffic between RabbitMQ server nodes

    RabbitMQ CLI communication with RabbitMQ server nodes

    RabbitMQ Management API and UI (backend connection from HAProxy to RabbitMQ)

```

Reply:
Thanks for pointing that out. I got the `kombu_ssl` names from the official OpenStack Security Guide[^1]. Seems like it is outdated in this regard.

I adjusted the section to specify the new `ssl*` options more precisely with a config snippet like you did.


[^1]: https://docs.openstack.org/security-guide/messaging/security.html#openstack-service-configuration-rabbitmq